lib/rules/declaration-property-max-values/index.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+const matchesStringOrRegExp = require('../../utils/matchesStringOrRegExp');
+const report = require('../../utils/report');
+const ruleMessages = require('../../utils/ruleMessages');
+const valueParser = require('postcss-value-parser');
+const vendor = require('../../utils/vendor');
+
+const ruleName = 'declaration-property-max-values';
+
+const messages = ruleMessages(ruleName, {
+	rejected: (property, max) =>
+		`Expected "${property}" to have no more than ${max} ${max === 1 ? 'value' : 'values'}`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/list/declaration-property-max-values',
+};
+
+/** @type {import('stylelint').Rule<Record<string, Number>>} */
+const rule = (primary) => {
+	return (root, result) => {
+		root.walkDecls((decl) => {
+			const prop = decl.prop;
+			const value = decl.value;
```

Comment:
[trivial] What about the following refactoring? (minor, though)

```suggestion
			const { prop, value } = decl;
```